### PR TITLE
Consolidate playbook for responsible tech choices

### PR DESCRIPTION
**Pull request by joeyAghion**
Our recent engineering skills survey revealed a pattern of problems with how Artsy makes technology choices, including low and/or decreasing scores on questions like:
* _I understand how Artsy Engineering makes choices about technology_
* _Artsy makes Quality Worthy of Art technology choices_
* _I understand how to select an appropriate technology stack for a new project_
* _I understand when it is appropriate to create a new service vs. adding to an existing one_

I revisited our relevant playbook searching for ways to remedy this and it had a number of obvious problems. Some of the "choices" it documents are out-of-date, and it nods to 2 completely different vocabularies for the stages of evaluating technology.

This PR tries to consolidate on the tech "radar" framework and adopt its hold/assess/trial/adopt language. I scrapped our old list of preferences and migrated the few that were still applicable into the radar data itself, so there's a single reference. (I don't expect any of these to be controversial, but any and all feedback is welcome.)

I added recommendations for "technical plans" and "technology review" discussions as tools in evolving tech choices through these stages. I mentioned some of the heuristics we've already used internally, like "don't experiment with too many things on a single system," "target a timeline when trialing a new tech," and "a critical mass of engineers should be familiar before something can be recommended broadly." Again, I'm aiming to put the unwritten into writing more than introduce any new governance.

Finally, there's a brief FAQ with real-world examples of each of the stage transitions.

Throughout this exercise, I tried to emphasize the positive.  (We truly _can_ choose the best/new tech as long as we take responsibility for past choices and necessary migrations or maintenance). In the distant past we didn't even pause to consider these things, and if we've been more conservative or negative recently it may be due to whiplash from that. In any case, I would love for this updated doc to be a cleaner and more consistent basis for future refinements.

### CODE REVIEW

File: playbooks/technology-choices.md [hunk 1]
```diff
@@ -6,106 +6,76 @@ description: Evaluating and adopting new technology at Artsy
...
+& learn or practice meeting discussion to review your experience and share any lessons. Make a pull request to the
+radar and make sure to request comments from the relevant engineers or experts. Remember that it may not be
+sufficient to just "adopt" a new choice. If this replaces an alternative that's in place at Artsy, that should
+probably move to "hold" and a strategy be decided for migrating away (e.g., opportunistically or deliberately).
+
+#### I just want to build a feature _this_ way or with _this_ library. Is a technical plan necessary?
+
+Use your judgment. Minor dependency selections may not warrant broad input. If a library or approach influences how
+future code will be written or how other developers will work, though, it's worth a time-out to get feedback,
+consider competing options, and choose deliberately.
```
Comment (ashfurrow) [inline playbooks/technology-choices.md]: I think it's worth noting here how the document is valuable as a historical/architecture decision record, even without discussion.

File: playbooks/technology-choices.md [hunk 2]
```diff
@@ -6,106 +6,77 @@ description: Evaluating and adopting new technology at Artsy
...
+
+We want to accomplish a lot with a lean team, which means we must
+[choose stable technologies](http://boringtechnology.club/). However, we also want to adopt best-of-breed
+technologies or best-suited tools to a given need, which may need work or still be evolving. We've borrowed from
+[ThoughtWorks' Radar](https://www.thoughtworks.com/radar/faq) to define the following stages for evaluating,
+adopting, and retiring technologies:
+
+- **Adopt:** Reasonable defaults for most work. These choices have been exercised successfully in production at
+  Artsy and there is a critical mass of engineers comfortable working with them.
+- **Trial:** These are being evaluated in limited production circumstances. We don't have enough production
```
Comment (dleve123) [inline playbooks/technology-choices.md]:
```suggestion
- **Trial:** These technologies are being evaluated in limited production circumstances. We don't have enough production
```

File: playbooks/technology-choices.md [hunk 3]
```diff
@@ -6,106 +6,77 @@ description: Evaluating and adopting new technology at Artsy
...
+  experience to recommend them for high-risk or business-critical use cases, but they may be worth consideration if
+  your project seems like a fit.
+- **Assess:** Technologies we are interested in and maybe even built proofs-of-concept for, but haven't yet trialed
+  in production.
+- **Hold:** Based on our experience, these technologies should be avoided. We've found them to be flawed, immature,
+  or simply supplanted by better alternatives. In some cases these remain in legacy production uses, but we should
+  take every opportunity to retire or migrate away.
+
+Artsy's current choices can be edited in [technology_radar/data.csv](/playbooks/technology_radar/data.csv) and
+[viewed in radar format here](https://radar.thoughtworks.com/?sheetId=https%3A%2F%2Fraw.githubusercontent.com%2Fartsy%2Freadme%2Fmaster%2Fplaybooks%2Ftechnology_radar%2Fdata.csv).
```
Comment (dleve123) [inline playbooks/technology-choices.md]: total nitpick, but visiting this link displays

Screen Shot 2019-12-10 at 2 36 09 PM

(notice the `DATA` header).

I wonder if renaming data.csv to artsy_technology_radar.csv would look a bit better / be more shareable.

File: playbooks/technology-choices.md [hunk 4]
```diff
@@ -6,106 +6,77 @@ description: Evaluating and adopting new technology at Artsy
...
+agreement, you may want to revisit the problem and rationale (or just ask engineering leadership to weigh in).
+
+When a plan depends on technology that isn't in the "adopt" category, it's worth special care to:
+
+- justify the suitability of the choice
+- document the goals of the "trial"
+- limit any production risk or organizational burden
+- if successful, plan to deprecate competing or overlapping technologies
+
+## Frequently Asked Questions
```
Comment (dleve123) [inline playbooks/technology-choices.md]: Love the FAQ!

File: playbooks/technology-choices.md [hunk 5]
```diff
@@ -6,106 +6,77 @@ description: Evaluating and adopting new technology at Artsy
...
-- **Focused services** over monolithic projects to ease developers' cognitive overhead, speed development and
-  testing, and allow for flexible technology evolution (see above). Projects can be focused around a _function_
-  (such as messaging) or a _domain_ (such as consignments).
-- **Focused, smaller management utilities** over earlier iterations of admin utilities such as
-  [Torque](https://admin.artsy.net). Examples include [CMS](https://cms.artsy.net)
-  ([Volt](https://github.com/artsy/volt)) for partners, [Helix](https://helix.artsy.net) for genomers, and
-  [auctions.artsy.net](https://auctions.artsy.net) ([Ohm](https://github.com/artsy/ohm)) for managing sales. We aim
-  to _opportunitistically_ transition remaining administrative functions (such as for inquiries or users) to new
-  tools.
-- **Opaque IDs** over "slugs" for any programmatic access. While slugs are more friendly for URLs, they can change
```
Comment (dleve123) [inline playbooks/technology-choices.md]: This is interesting as it is something that doesn't totally fit into the model of a radar (or perhaps I don't see it).

How do you think we should communicate out these technology strategies given the approach of this PR? Shall we migrate to the radar?

File: playbooks/technology_radar/data.csv [hunk 6]
```diff
@@ -1,29 +1,32 @@
...
 Typescript,adopt,languages & frameworks,FALSE,
-UIKit,adopt,languages & frameworks,FALSE,
+UIKit,adopt,languages & frameworks,FALSE,"See Objective-C."
 AWS,adopt,platforms,FALSE,
 Docker,adopt,platforms,FALSE,
 HSTS,adopt,platforms,FALSE,
-Kubernetes,adopt,platforms,FALSE,
+Kubernetes,adopt,platforms,FALSE,"We manage staging and production clusters, and have solutions for logging, monitoring, alerting, scaling, and routing. See Hokusai and Substance repos."
 Redis,adopt,platforms,FALSE,
+Responsive web pages,adopt,techniques,FALSE,"Rather than mobile- or desktop-specific designs, we prefer pages to degrade gracefully."
```
Comment (dleve123) [inline playbooks/technology_radar/data.csv]: Bit related to an above comment, but thoughts on adding an entry for the Opaque Ids technique?

File: playbooks/technology-choices.md (hunk 5 above)
Comment (joeyAghion) [reply playbooks/technology-choices.md]: Yeah the radar's quadrants aren't always a good fit. I think it's still better to cram it into a category that might not be perfect anyway. At least it's documented and discussed that way. Will add this.

File: playbooks/technology-choices.md (hunk 3 above)
Comment (joeyAghion) [reply playbooks/technology-choices.md]: Weird. Not sure if this functionality changed or I just never noticed that before.

I'll rename in a separate PR to keep this PR's diff readable.

**Comment by joeyAghion** [comment]
Going for it! File rename coming up next...

**Pull request merged**